specs/0109-infobar-component.md

```diff
@@ -0,0 +1,58 @@
+* Feature Name: Infobar Component
+* Created: 2021-02-25
+* Associated Bugs: [bug 1694911](https://bugzilla.mozilla.org/show_bug.cgi?id=1694911)
+
+# Summary
+
+Create a reusable component to display infobars to users.
+
+# Motivation
+
+We need a flexible way to notify users without writing ad-hoc code for every situation:
+* Infobars should be manageable by admins without intervention from developers.
+* Unlike notifications, infobars are more visible and require interaction from users to get dismissed.
+
+Example use case: we introduce a brand new feature, like string comments, and we want to make sure that users are aware of it. The infobar will include a short message and a link to documentation.
+
+# Feature explanation
+
+When a logged in user opens Pontoon, we check against a list of infobars. If the user matches the necessary conditions, we display the infobar.
+
+Information about each infobar is stored in a table with the following fields:
+* Identifier: short text identifier, e.g. `new_terms_2020`.
+* Start time (datetime): when we should start displaying the infobar to users.
+* End time (datetime): when we should stop displaying the infobar to users.
+* Creation date (date): stores when the infobar was created.
+* Cohort: if defined, we only display the infobar for users part of a specific experiment or cohort (A/B testing).
+* Type: `infobar`, `modal`. Infobar is displayed at the top of the window, while modal is displayed in the middle of the window.
```

Comment:
Seems like an infobar can be an infobar or modal. Can we come up with better names? Maybe "announcement" could be of type infobar or modal.

Also, what's the use case for modal now that we dropped terms update?

Reply:
Uh, that's a very good point. The only use case I had in mind for "modal" was the terms update. Should we just get rid of this? It would definitely simplifies things.

Reply:
If we don't have a use case for it, we should.